### ISSUE TITLE
Add an markup extension to bind standard values of a type

### ISSUE THREAD

**Issue by YohDeadfall** [dotnet/wpf#655]
A long time ago I wrote [Enumerate](https://github.com/YohDeadfall/Enumerate) which is designed to provide standard values of any type if it supports it. It's used mostly for `enum`s, but as I said it isn't limited to them. And because WPF is open source now, I think it's time to make the markup extension built-in so everyone will be able to use out of the box.

## Features

1. Providing values of an `enum` as is.
2. Providing values of an `enum` from `DescriptionAttribute`s.
1. Providing values of any type whose `TypeConverter` returns the standard values via the `GetStandardValues` method.

## Usage:

```csharp

public enum ViewModelStatus
{
    [Description("Offen")]
    Open,
    [Description("Geschlossen")]
    Closed,
    [Description("In Arbeit")]
    InProgress
}

public class ViewModel : INotifyPropertyChanged
{
    public ViewModelStatus Status { get; set; }
}

```

```xaml
<ComboBox ItemsSource="{e:Enumerate {x:Type local:ViewModelStatus}, Converter={StaticResource EnumToStringConverter}}"
          SelectedItem="{Binding Status, Converter={StaticResource EnumToStringConverter}}" />
<!-- ItemsSource will contain Offen, Geschlossen and In Arbeit -->
```

---

Related to #455.

**Comment by ryalanms** [comment]
Thanks, @YohDeadfall.

**Comment by miloush** [comment]
I agree that using an enum as items source seems to be a common enough scenario to be considered for a built-in support. However, I believe there should be some discussion and a proper specification of the feature.

I value the nullability and localization support in your implementation, however the localization approach for example doesn't fit well within the WPF localization model. I would also prefer a strong typed model to converters, and it might be worth to cache the values for performance reasons.
